## reproduction steps

using Scala 2.13.4 and scalameta 4.3.24,

```scala
import scala.meta._

object Test extends App {
  "".parse[Source] match {
    case Parsed.Success(_) => println("Success")
    case Parsed.Error(_, _, _) => println("Error")
  }
}
```

## problem

The Scala 2.13.4 compiler produces the following message:

```
.../Test.scala:4:11: match may not be exhaustive.
[warn] It would fail on the following inputs: Error(), Success()
[warn]   "".parse[Source] match {
[warn]           ^
[warn] one warning found
[info] done compiling
```

The Scala 2.13.3 compiler does not produce this warning.

This might be related to issue #12186. 

Attached is a zip file to ease reproducing the issue. 
[scala-2.13.4-match-issue.zip](https://github.com/scala/bug/files/5567783/scala-2.13.4-match-issue.zip)
